scripts/pre_commit_linter.py

```diff
@@ -1463,6 +1471,106 @@ def _check_directive_scope(all_files):
+                    continue
+                property_name = expression['callee']['property']['name']
+                if property_name not in properties_to_check:
+                    continue
+                arguments = expression['arguments']
+                if arguments[0]['type'] == 'Literal':
+                    property_value = str(arguments[0]['value'])
+                arguments = arguments[1:]
+                for argument in arguments:
+                    if argument['type'] != 'ArrayExpression':
+                        continue
+                    literal_args = []
+                    function_args = []
+                    dollar_imports = []
+                    regular_imports = []
+                    constant_imports = []
+                    elements = argument['elements']
+                    for element in elements:
+                        if element['type'] == 'Literal':
+                            literal_args.append(str(element['value']))
+                        elif element['type'] == 'FunctionExpression':
+                            func_args = element['params']
+                            for func_arg in func_args:
+                                function_args.append(str(func_arg['name']))
+                    for arg in function_args:
+                        if arg.startswith('$'):
+                            dollar_imports.append(arg)
+                        elif re.search('[a-z]', arg):
+                            regular_imports.append(arg)
+                        else:
+                            constant_imports.append(arg)
+                    dollar_imports.sort()
+                    regular_imports.sort()
+                    constant_imports.sort()
+                    sorted_imports = (
+                        dollar_imports + regular_imports + constant_imports)
+                    if sorted_imports != function_args:
+                        failed = True
+                        print (
+                            'Please ensure that %s in %s, the injected '
+                            'dependecies should be in the following manner:'
+                            ' dollar imports, regular imports and constant '
+                            'imports, all in sorted form.'
+                            % (property_value, filename))
+                    if sorted_imports != literal_args:
+                        failed = True
+                        print (
+       	                    'Please ensure that %s in %s, the dependecies '
+       	                    'literally mentioned should be in the following '
+       	                    ' manner: dollar imports, regular imports and '
+       	                    'constant imports, all in sorted form.'
+       	                    % (property_value, filename))
+
+    if failed:
+        summary_message = (
+            '%s   Sorted dependencies check failed' % (
+                _MESSAGE_TYPE_FAILED))
+        print summary_message
```

Comment:
Pull this and the next line out: they are being repeated in both the blocks.

Reply:
Done